hca_ingest/downloader/data_collector.py

```diff
@@ -88,22 +100,46 @@ def __set_inputs(entity_dict, linking_map):
                 try:
                     protocols = [entity_dict[protocol_id] for protocol_id in protocol_ids]
                 except Exception as e:
-                    raise RuntimeError(f'problem with process {process_id} and protocol: {str(e)}, for  entity {entity}') from e
+                    raise RuntimeError(
+                        f'problem with process {process_id} and protocol: {str(e)}, for  entity {entity}') from e
                 try:
                     input_biomaterials = [entity_dict[id] for id in input_biomaterial_ids]
                 except Exception as e:
-                    raise RuntimeError(f'problem with process {process_id} and biomaterial: {str(e)}, for  entity {entity}') from e
+                    raise RuntimeError(
+                        f'problem with process {process_id} and biomaterial: {str(e)}, for  entity {entity}') from e
                 try:
                     input_files = [entity_dict[id] for id in input_files_ids]
                 except Exception as e:
-                    raise RuntimeError(f'problem with process {process_id} and file: {str(e)}, for  entity {entity}') from e
+                    raise RuntimeError(
+                        f'problem with process {process_id} and file: {str(e)}, for  entity {entity}') from e
 
                 entity.set_input(input_biomaterials, input_files, process, protocols)
 
-    def __get_entities_by_submission_and_type(self, data_by_submission, submission, entity_type):
-        self.api.page_size = 1000
-        entity_json = \
-            self.api.get_related_entities(entity_type, submission, entity_type)
-        self.api.page_size = 100
-        if entity_json:
-            data_by_submission.extend(list(entity_json))
+    def __get_entities_by_submission_and_type(self, submission, entity_type):
+        self.api.page_size = 10
+        yield from self.api.get_related_entities(entity_type, submission, entity_type)
```

Comment:
I believe the tweaking of page size is only a temporary solution.
This is probably where we would need to use a Mongo Aggregation Framework query.